website/static/js/profile.js

```diff
@@ -446,6 +456,32 @@ var NameViewModel = function(urls, modes, preventUnsaved, fetchCallback) {
     self.fetch(fetchCallback);
 };
 NameViewModel.prototype = Object.create(BaseViewModel.prototype);
+
+NameViewModel.prototype.submit = function() {
+    if (this.hasValidProperty() && this.isValid()) {
+        // Name on Name view is empty
+        if (this.nameFieldEmpty()) {
```

Comment:
This shouldn't be necessary--can you use the HTML `required` attribute on the name field instead?